src/bitcoin-cli.cpp

```diff
@@ -947,7 +947,8 @@ static void ParseError(const UniValue& error, std::string& strPrint, int& nRet)
             strPrint += ("error message:\n" + err_msg.get_str());
         }
         if (err_code.isNum() && err_code.getInt<int>() == RPC_WALLET_NOT_SPECIFIED) {
-            strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to bitcoin-cli command line.";
+            strPrint += " Or for the CLI, specify the \"-rpcwallet=<filename>\" option before the command";
```

Comment:
I found "filename" confusing, see #30912 - if there isn't a deeper reason behind it that I'm unaware of, it could be renamed to `walletname`.

Reply:
I think it would be beneficial to explain the alternatives. The `-rpcwallet` parameter depends on how the wallet is loaded. It can either be the wallet directory name located inside the `-walletdir` (the default value is `datadir/net/wallets/wallet_name`, in which case only `wallet_name` is used in this option), or the absolute path to the wallet directory, which is the complete path `datadir/net/wallets/wallet_name`.

You can note the difference by loading an existing wallet in two different ways:
1) `loadwallet "test_wallet"`
2) `loadwallet "datadir/regtest/wallets/test_wallet"`

Then, call `getwalletinfo()`. The wallet name will differ, and as a result, the `-rpcwallet=<arg>` argument for the very same wallet will be different as well.

Reply:
... and I think in both of these cases `walletname` would be correct and `filename` incorrect, it would just be a different wallet name depending on how it's loaded.

It could actually also be a filename: Put a legacy wallet `oldwallet.dat` into the datadir, run with `-deprecatedrpc=create_bdb` and `loadwallet "oldwallet.dat"` will load it successfully, in this special case the filename is the walletname. But even here, where `filename` would make sense, `walletname` is correct too, so it just seem better overall.